p2p/pex/pex_reactor.go

```diff
@@ -372,6 +371,48 @@ func (r *PEXReactor) ensurePeers() {
 	}
 }
 
+func (r *PEXReactor) dialPeer(addr *p2p.NetAddress) {
+	var attempts int
+	var lastDialed time.Time
+	if lAttempts, attempted := r.attemptsToDial.Load(addr.DialString()); attempted {
+		attempts = lAttempts.(_attemptsToDial).number
+		lastDialed = lAttempts.(_attemptsToDial).lastDialed
+	}
+
+	if attempts > maxAttemptsToDial {
```

Comment:
Looks like we do have a maxAttempts mechanism already in place: https://github.com/tendermint/tendermint/blob/develop/p2p/pex/known_address.go#L130

Maybe we can consolidate to make sure they're working right together ? feel free to leave this an just open an issue to note the redundancy